### ISSUE TITLE
"Wrong number of argument" errors in REPL after cider-jack-in

### ISSUE THREAD

**Issue by seanredmond** [clojure-emacs/cider#935]
After M-x cider-jack-in, the REPL seems to start up fine, but as soon as I start typing in that buffer I see errors (and nothing ever gets evaluated). For instance, after just typing "(" I get (in \*Messages\*)
```
eldoc error: (wrong-number-of-arguments (lambda nil The connection to use for nREPL interaction. (or nrepl-connection-dispatch nrepl-connection-buffer (car (nrepl-connection-buffers)))) 1)
```
and the debugging output with toggle-debug-on-error:

```
Debugger entered--Lisp error: (wrong-number-of-arguments (lambda nil "The connection to use for nREPL interaction." (or nrepl-connection-dispatch nrepl-connection-buffer (car (nrepl-connection-buffers)))) 1)
  nrepl-current-connection-buffer(no-error)
  cider-connected-p()
  cider-eldoc()
  #[0 "\302 \204

**Comment by bbatsov** [comment]
CIDER and Emacs version info?

**Comment by seanredmond** [comment]
Argh, a whole bunch got cut off because of funky chars in the debugging output, though the preview seemed to handle it fine.

GNU Emacs 24.4.1 (x86_64-apple-darwin13.4.0, NS apple-appkit-1265.21)
Cider 0.8.2

This sounds similar to #492 although the version is newer. Trying again with the debugging output:
```
Debugger entered--Lisp error: (wrong-number-of-arguments (lambda nil "The connection to use for nREPL interaction." (or nrepl-connection-dispatch nrepl-connection-buffer (car (nrepl-connection-buffers)))) 1)
  nrepl-current-connection-buffer(no-error)
  cider-connected-p()
  cider-eldoc()
  #[0 "\302 \204\205G\303\304!\210\304\207	\203\303	 !\207\305 \306 \211\204$\304\202B@=\2038\307\310\"\206B\311!\202B\311!\206B\307\310\"\303!\266\203\207" [eldoc-last-message eldoc-documentation-function eldoc-display-message-p eldoc-message nil eldoc-current-symbol eldoc-fnsym-in-current-sexp apply eldoc-get-fnsym-args-string eldoc-get-var-docstring] 5 "\n\n(fn)"]()
  funcall(#[0 "\302 \204^0^@^H\205G^@\303\304!\210\304\207	\203^X^@\303	 !\207\305 \306 \211\204$^@\304\202B^@^A^A@=\2038^@\307\310^B\"\206B^@\311^B!\202B^@\311^B!\206B^@\307\310^B\"\303^A!\266\203\207" [eldoc-last-message eldoc-documentation-function eldoc-display-message-p eldoc-message nil eldoc-current-symbol eldoc-fnsym-in-current-sexp apply eldoc-get-fnsym-args-string eldoc-get-var-docstring] 5 "\n\n(fn)"])
  eldoc-print-current-symbol-info()
  #[0 "^H\205^F^@301 \207" [eldoc-mode eldoc-print-current-symbol-info] 1 "\n\n(fn)"]()
  apply(#[0 "^H\205^F^@\301 \207" [eldoc-mode eldoc-print-current-symbol-info] 1 "\n\n(fn)"] nil)
  byte-code("r\301^H302H^H\303H\"\210)\301\207" [timer apply 5 6] 4)
  timer-event-handler([t 0 0 500000 t #[0 "^H\205^F^@\301 \207" [eldoc-mode eldoc-print-current-symbol-info] 1 "\n\n(fn)"] nil idle 0])

```

**Comment by seanredmond** [comment]
After looking more at comments on #492 and similar issues and digging into my packages, I figured out that I had some other, conflicting nrepl package installed (from a long time ago and I think before I was using melpa-stable). Just uninstalling that package fixed this problem.

**Comment by bbatsov** [comment]
Glad to hear you solved this.

On Friday, January 2, 2015, Sean Redmond <notifications@github.com> wrote:

> Closed #935 <https://github.com/clojure-emacs/cider/issues/935>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/clojure-emacs/cider/issues/935#event-213359096>.
>


-- 
Best Regards,
Bozhidar Batsov

http://www.batsov.com
